File: 12_power.py
# -*- coding:utf-8 -*-
class Solution:
    def pow(self, base, exponent):
        tmp = 1
        while(exponent > 0):
            tmp *= base
            exponent -= 1
        return tmp
    def Power(self, base, exponent):
        # write code here
        if base == 0: return 0
        if exponent == 0: return 1
        if exponent < 0:
            res = self.pow(base, -1 * exponent)
            return 1.0 / res
        return self.pow(base, exponent)

solution = Solution()
base = 2
exponent = 3
print(solution.Power(base, exponent))

File: 12_power.cpp
class Solution {
public:
    double Power(double base, int exponent) {
        if(exponent == 0) return 1;
        double res = 1;
        if(exponent > 0)
        {
            for(int i = 1; i <= exponent; ++i)
            {
                res *= base;
            }
            return res;
        }
        
        else
        {
            for(int i = 1; i <= -1 * exponent; ++i)
            {
                res *= base;
            }
            return 1 / res;
        }
    }
};


// 给定一个double类型的浮点数base和int类型的整数exponent。
// 求base的exponent次方。

// 运行时间：3ms

// 占用内存：476k